### PR TITLE
azure pipelines cffi wheel builder

### PR DESCRIPTION
**Pull request by reaperhulk**
it works!

### CODE REVIEW

File: external-projects/cffi-wheel-builder.yml [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+          vmImage: 'macOS-10.14'
+      strategy:
+          matrix:
+              Python27:
+                  PYTHON_DOWNLOAD_URL: "https://www.python.org/ftp/python/2.7.16/python-2.7.16-macosx10.6.pkg"
+                  PYTHON_BIN_PATH: /Library/Frameworks/Python.framework/Versions/2.7/bin/python
+              Python34:
+                  PYTHON_DOWNLOAD_URL: "https://www.python.org/ftp/python/3.4.4/python-3.4.4-macosx10.6.pkg"
+                  PYTHON_BIN_PATH: /Library/Frameworks/Python.framework/Versions/3.4/bin/python3
+              Python35:
```
Comment (alex) [inline external-projects/cffi-wheel-builder.yml]: Remind me why these aren't abi3 wheels?

File: external-projects/cffi-wheel-builder.yml [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+          - script: |
+              set -e
+              set -x
+              # This is done via get-pip because Python 3.4/3.5 are linked against
+              # an ancient OpenSSL so we have no TLS 1.2 unless we upgrade pip
+              # (which uses the SecureTransport ctypes shim)
+              curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
+              # pinning pip because pip > 19.1 drops 3.4 support
+              $PYTHON_BIN_PATH get-pip.py 'pip==19.1.1'
```
Comment (alex) [inline external-projects/cffi-wheel-builder.yml]: Wont' `get-pip.py` respect `requires_python`?

File: external-projects/cffi-wheel-builder.yml (hunk 1 above)
Comment (reaperhulk) [reply external-projects/cffi-wheel-builder.yml]: cffi can’t do abi3. It needs access to the guts.

File: external-projects/cffi-wheel-builder.yml (hunk 2 above)
Comment (reaperhulk) [reply external-projects/cffi-wheel-builder.yml]: Probably does?

Comment (alex) [reply external-projects/cffi-wheel-builder.yml]: Then we don't need the explicit version

File: external-projects/cffi-wheel-builder.yml [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+            displayName: Build the wheel
+          - script: .venv/bin/pip install --no-index -f wheelhouse cffi
+            displayName: Test installing the wheel
+          - script: |
+              .venv/bin/python -c "import cffi;print(cffi.__version__)"
+            displayName: Print the cffi version
+          - task: PublishBuildArtifacts@1
+            inputs:
+                pathToPublish: wheelhouse/
+                artifactName: cffi-macos-python$(python.version)
```
Comment (alex) [inline external-projects/cffi-wheel-builder.yml]: python.version is not set, you need to interpolate something different

Comment (alex) [reply external-projects/cffi-wheel-builder.yml]: (The result of this is that only a single macOS wheel is uploaded: https://dev.azure.com/pyca/infra/_build/results?buildId=414&view=results)

**Pull request merged**